jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -269,13 +273,36 @@ def _on_flux_unit_changed(self, msg):
         else:
             self.flux_or_sb_selected = 'Surface Brightness'
 
+        # Always send a surface brightness unit to contours
+        if self.flux_or_sb_selected == 'Flux':
+            yunit = self._append_angle_correctly(yunit, self.angle_unit.selected)
+        self._find_and_convert_contour_units(yunit)
+
         # for displaying message that PIXAR_SR = 1 if it is not found in the FITS header
         if (
             len(self.app.data_collection) > 0
             and not self.app.data_collection[0].meta.get('PIXAR_SR')
         ):
             self.pixar_sr_exists = False
 
+    def _find_and_convert_contour_units(self, yunit=None):
```

Comment:
would it be hard to have this take `viewer` as an optional argument so that calls from the `AddToViewerMessage` only act on the single applicable viewer instead of looping over all viewers?

Reply:
Hmm, but isn't unit conversion global, and would thus apply to contours in another viewer as well?

Reply:
yes, but in theory when `AddDataToViewerMessage` is calling this, only _that_ viewer/layer combo needs to be updated, right?